### PR TITLE
Add PATCH E2E test coverage aligned with PUT tests

### PR DESCRIPTION
**Pull request by rbans96**
## Description
PATCH operation tests are added covering the same scenarios and edge cases currently validated by PUT tests

## Related issues
Addresses https://microsofthealth.visualstudio.com/Health/_workitems/edit/178574

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- When changing or adding behavior, if your code modifies the system design or changes design assumptions, please create and include an [ADR](https://github.com/microsoft/fhir-server/blob/main/docs/arch).
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/main/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/main/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

**Review by rbans96 — COMMENTED** [review]
##  PR Review Assistant

### Summary


### Risk Assessment:  Low (/10)
> 

** Files Changed (Low-Risk):**
  - `test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/FhirPathPatchTests.cs`
  - `test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/JsonPatchTests.cs`

###  No blocking issues found

---
###  Would Auto-Approve
No high-risk files changed  AI risk score /10  3  Only E2E test files modified.
> *Note: GitHub does not allow self-approval  a different reviewer must approve.*

*Powered by [PR Review Assistant Agent](https://github.com/microsoft/fhir-server/blob/agents/pr-review-assistant-agent/docs/pr-review-agent.md)  GitHub Models gpt-4o-mini*

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/JsonPatchTests.cs [hunk 1]
```diff
@@ -457,5 +459,76 @@ public async Task GivenAnEmptyJsonPatchBody_WhenPatchJsonEndpointIsCalled_ThenSe
...
+            Assert.Equal(originalVersionId, patchResponse.Resource.Meta.VersionId);
+            Assert.Equal(originalLastUpdated, patchResponse.Resource.Meta.LastUpdated);
+        }
+
+        /// <summary>
+        /// Tests that patching a single field does not inadvertently modify other resource fields, verifying the surgical precision of JSON PATCH.
+        /// </summary>
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAnExistingPatient_WhenJsonPatchingOneField_ThenOtherFieldsShouldNotBeModified()
```
Comment (apurvabhaleMS) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/JsonPatchTests.cs:515]: Kind of partially covered by existing test GivenAServerThatSupportsIt_WhenSubmittingALevel1PropertyPatch_ThenServerShouldPatchCorrectly

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/FhirPathPatchTests.cs [hunk 2]
```diff
@@ -706,5 +707,77 @@ public async Task GivenPatchFhirRequestWithEmptyBody_WhenInvoked_ThenReturnsBadR
...
+            Assert.Equal(originalVersionId, patchResponse.Resource.Meta.VersionId);
+            Assert.Equal(originalLastUpdated, patchResponse.Resource.Meta.LastUpdated);
+        }
+
+        /// <summary>
+        /// Tests that patching a single field does not inadvertently modify other resource fields, verifying the surgical precision of FHIR path PATCH.
+        /// </summary>
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAnExistingPatient_WhenFhirPatchingOneField_ThenOtherFieldsShouldNotBeModified()
```
Comment (apurvabhaleMS) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/FhirPathPatchTests.cs:764]: Kind of partially covered by existing test GivenAServerThatSupportsIt_WhenSubmittingASimplePropertyPatch_ThenServerShouldPatchCorrectly

**Review by apurvabhaleMS — APPROVED** [review]

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/JsonPatchTests.cs (hunk 1 above)
Comment (rbans96) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/JsonPatchTests.cs:515]: Thanks for the review! You're right that these tests share some overlap with the existing simple property patch tests.

The distinction is: the existing tests patch multiple fields (e.g. replace gender + delete address) and verify all those changes applied. The new tests patch only a single field and explicitly assert that untouched fields remain unchanged (address count, resource ID). This directly addresses acceptance criteria: "Tests verify that PATCH updates only the intended fields and do not unintentionally modify other resource data."

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/FhirPathPatchTests.cs (hunk 2 above)
Comment (rbans96) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/FhirPathPatchTests.cs:764]: same for this

**Review by rbans96 — COMMENTED** [review]
## Review 🟢 3/10

This PR adds tests for handling invalid ETag formats and ensures that patching does not create new versions when there are no data changes.

> The changes primarily add tests, but there are gaps in coverage for error scenarios.

⚠️ — `test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/FhirPathPatchTests.cs` (line 707) Missing test for the scenario where the resource does not exist when patching.
⚠️ — `test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/FhirPathPatchTests.cs` (line 707) Missing test for the scenario where the If-Match header is not provided.
⚠️ — `test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/JsonPatchTests.cs` (line 459) Missing test for the scenario where the resource does not exist when applying a JSON patch.

*Not auto-approved: Failing CI checks: Check Metadata, Check Metadata, Check Metadata*
